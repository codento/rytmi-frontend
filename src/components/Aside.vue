<template>
  <aside class="aside-menu" />
</template>

<script>
export default {
  name: 'c-aside'
}
</script>
