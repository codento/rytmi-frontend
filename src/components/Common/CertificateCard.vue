<template>
  <b-card
    class="my-2 education-preview"
    no-header
    no-body
  >
    <b-card-body>
      <b-card-text>
        <h4>{{ certificateData[language].name }} <span class="small">{{ certificateData.year }}</span></h4>
        <p> {{ certificateData[language].description }} </p>
      </b-card-text>
    </b-card-body>
  </b-card>
</template>

<script>
import { LANGUAGE_ENUM } from '@/utils/constants'

export default {
  name: 'CertificateCard',
  props: {
    certificateData: {
      type: Object,
      required: true
    },
    language: {
      type: Text.language,
      required: true,
      validator: (value) => {
        return LANGUAGE_ENUM.LANGUAGES.map(item => item.id).indexOf(value) !== -1
      }
    }
  }
}
</script>

<style scoped >
.certificate-preview > h4 {
  font-weight: bolder;
}
</style>
