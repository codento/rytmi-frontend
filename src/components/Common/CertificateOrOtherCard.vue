<template>
  <b-card
    class="my-2 certificate-or-other-preview"
    no-header
    no-body
  >
    <b-card-body>
      <b-card-text>
        <h4>{{ certificateOrOther[language].name }} <span class="small">{{ certificateOrOther.year }}</span></h4>
        <p> {{ certificateOrOther[language].description }} </p>
      </b-card-text>
    </b-card-body>
  </b-card>
</template>

<script>
import { LANGUAGE_ENUM } from '@/utils/constants'

export default {
  name: 'CertificateOrOtherCard',
  props: {
    certificateOrOther: {
      type: Object,
      required: true
    },
    language: {
      type: Text.language,
      required: true,
      validator: (value) => {
        return LANGUAGE_ENUM.LANGUAGES.map(item => item.id).indexOf(value) !== -1
      }
    }
  }
}
</script>

<style scoped >
.certificate-or-other-preview > h4 {
  font-weight: bolder;
}
</style>
