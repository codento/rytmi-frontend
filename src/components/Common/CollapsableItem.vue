<template>
  <div>
    <b-card
      no-body
      class="mb-1 rounded"
    >
      <b-card-header
        class="p-1 btn-rytmi-light"
      >
        <b-button
          block
          variant="rytmi-light"
          @click="visibility = !visibility"
        >
          {{ title }}
          <span>
            <i
              :class="visibility ? 'fa fa-chevron-up' : 'fa fa-chevron-down'"
            />
          </span>
        </b-button>
      </b-card-header>
      <b-collapse
        id="collapse"
        v-model="visibility"
      >
        <slot />
      </b-collapse>
    </b-card>
  </div>
</template>

<script>
export default {
  name: 'CollapsableItem',
  props: {
    title: String,
    initialVisibility: {
      type: Boolean,
      default: false
    }
  },
  data () {
    return {
      visibility: this.initialVisibility
    }
  }
}
</script>
<style scoped >
  .btn {
    font-size: 18px;
    font-weight: 500;
  }
</style>
