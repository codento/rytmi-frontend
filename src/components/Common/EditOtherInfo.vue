<template>
  <b-row class="mt-2">
    <b-col>
      <b-tabs content-class="mt-1 border-0">
        <b-tab
          active
          title="Edit"
        >
          <label
            v-if="inputLabel.length > 0"
            for="input-other-info"
          >
            {{ inputLabel }}
          </label>
          <b-textarea
            id="input-other-info"
            v-model="otherInfoAsMarkdown"
            :rows="rows"
            :placeholder="placeholderText"
            :state="isValidMarkdown"
            @input="updateMarkdown"
          />
          <b-form-invalid-feedback
            id="input-other-info-feedback"
            class="text-left"
          >
            {{ errorMessage }}
          </b-form-invalid-feedback>
        </b-tab>
        <b-tab
          title="Template"
        >
          <b-textarea
            :rows="rows"
            :value="placeholderText"
            plaintext
          />
        </b-tab>
      </b-tabs>
    </b-col>
    <b-col>
      <div v-html="compiledMarkdown" />
    </b-col>
  </b-row>
</template>
<script>
import _ from 'lodash'
import marked from 'marked'

export default {
  name: 'EditOtherInfo',
  props: {
    languageKey: String,
    inputText: String,
    inputLabel: {
      type: String,
      default: ''
    },
    rows: {
      type: Number,
      default: 10
    }
  },
  data () {
    return {
      otherInfoAsMarkdown: '',
      placeHolders: {
        'fi': '## Koulutus\n#### Esimerkkioppilaitos, 2012\nJoku tutkinto, pääaine (sivuaine)',
        'en': '## Education\n#### Example university, 2012\nSome degree, major (minor)'
      },
      requiredItems: {
        'fi': [{ requiredItem: '## Koulutus', errorMessage: 'Education information missing, can\'t find markdown element ## Koulutus.' }],
        'en': [{ requiredItem: '## Education', errorMessage: 'Education information missing, can\'t find markdown element ## Education.' }]
      }
    }
  },
  computed: {
    compiledMarkdown: function () {
      return marked(this.otherInfoAsMarkdown, { sanitize: true })
    },
    placeholderText: function () {
      return this.placeHolders[this.languageKey]
    },
    isValidMarkdown: function () {
      return this.requiredItems[this.languageKey].every(a => this.otherInfoAsMarkdown.includes(a.requiredItem))
    },
    errorMessage: function () {
      const missingItems = this.requiredItems[this.languageKey].filter(a => !this.otherInfoAsMarkdown.includes(a.requiredItem))
      return missingItems.map(item => item.errorMessage).join(' ')
    }
  },
  mounted: function () {
    this.otherInfoAsMarkdown = this.inputText
  },
  methods: {
    updateMarkdown: function () {
      _.debounce(function (e) {
        this.otherInfoAsMarkdown = e.target.value
      }, 300)
      this.$emit('input-updated', this.otherInfoAsMarkdown, this.languageKey)
    }
  }
}
</script>
