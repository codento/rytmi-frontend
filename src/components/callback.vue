<template>
  
</template>

<script>
import {mapActions} from 'vuex'

export default {
  created () {
    this.$nextTick(() => {
      this.fetchProfiles()
      this.fetchSkills()
      this.fetchProfileSkills()
      this.$router.push(this.$route.query.redirect || '/')
    })
  },
  methods: {
    ...mapActions([
      'fetchProfiles',
      'fetchSkills',
      'fetchProfileSkills'
    ])
  }
}
</script>