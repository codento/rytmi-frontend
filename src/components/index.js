import Aside from './appStructures/Aside.vue'
import Breadcrumb from './appStructures/Breadcrumb.vue'
import Callout from './helpers/Callout.vue'
import Footer from './appStructures/Footer.vue'
import Header from './appStructures/Header.vue'
import Sidebar from './appStructures/Sidebar/Sidebar.vue'

export {
  Aside,
  Breadcrumb,
  Callout,
  Footer,
  Header,
  Sidebar
}
