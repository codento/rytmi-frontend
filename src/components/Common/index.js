import ProjectRow from './ProjectRow.vue'
import SkillRow from './SkillRow.vue'
import SkillExplanations from './SkillExplanations.vue'
import UtilizationChart from './UtilizationChart.vue'
import CollapsableItem from './CollapsableItem.vue'
import EducationCard from './EducationCard.vue'
import ProjectForm from './ProjectForm.vue'
import ProjectProfileForm from './ProjectProfileForm.vue'

export {
  ProjectRow,
  SkillRow,
  SkillExplanations,
  UtilizationChart,
  CollapsableItem,
  EducationCard,
  ProjectForm,
  ProjectProfileForm
}
