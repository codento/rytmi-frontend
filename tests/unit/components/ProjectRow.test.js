import { mount, createLocalVue } from '@vue/test-utils'
import { ProjectRow } from '@/components/Common'
import Vuex from 'vuex'
import { format } from 'date-fns'

const mockProject = {
  projectId: 1,
  name: { fi: 'Testi prkkis' },
  description: { fi: 'mail' },
  startDate: '2017-05-15',
  endDate: '2017-06-16'
}

describe('ProjectRow.test.js', () => {
  let localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.filter('dateFilter', value => {
    return value ? format(value, 'D.M.YYYY') : undefined
  })
  let getters, state, store
  beforeEach(() => {
    getters = {
      projectById: () => (arg) => mockProject,
      currentLanguage: () => 'fi'
    }
    state = {
      siteSettings: { currentLanguage: 'fi' }
    }
    store = new Vuex.Store({ getters, state })
  })
  it('Template is correct', () => {
    let wrapper = mount(ProjectRow, {
      store,
      localVue,
      propsData: {
        profileProject: mockProject
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
