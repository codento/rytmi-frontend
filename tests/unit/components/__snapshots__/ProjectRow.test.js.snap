// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectRow.test.js Template is correct 1`] = `
<div>
  <div name="show-row" class="row mb-1">
    <div class="col-sm-12">
      <router-link to="[object Object]" class="no-underline-router-link"><span class="project-header">Testi prkkis </span></router-link>
    </div>
    <div class="col-sm-12 duration-container"><span class="duration-item"><small>From</small><br>
        15.5.2017
      </span> <span class="duration-item"><small>To</small><br>
        16.6.2017
      </span></div>
    <!---->
  </div>
  <hr>
</div>
`;
